### ISSUE TITLE
Why apply "pred_fake = netD(fake.detach())" in the function "backward_D_basic" in models/architecture.py?

### ISSUE THREAD

**Issue by crissallan** [DeepMotionEditing/deep-motion-editing#90]
Hi, 

According to my understanding, if .detach() applied, the back-propagation will end here, and the gradients can't afftect the weights of the generator. Then this adverserial-training might not affect the performance of the generator at all.

Many thanks!

**Comment by PeizhuoLi** [comment]
In adversarial training, when optimizing the discriminator, it is necessary to apply detach() to fake result so that the gradient won't follow back to generator. However, when optimizing generator, we don't apply detach() to fake result.

**Comment by crissallan** [comment]
> In adversarial training, when optimizing the discriminator, it is necessary to apply detach() to fake result so that the gradient won't follow back to generator. However, when optimizing generator, we don't apply detach() to fake result.

Thanks!
